tasmota/tasmota_xsns_sensor/xsns_120_dfrobot_mics.ino

```diff
@@ -0,0 +1,276 @@
+  itoa(DFRobot_MICS_data.Ethanol_b, DFRobot_MICS_ch.Ethanol_b, 10);
+  itoa(DFRobot_MICS_data.Propane_b, DFRobot_MICS_ch.Propane_b, 10);
+  itoa(DFRobot_MICS_data.Iso_Butane_b, DFRobot_MICS_ch.Iso_Butane_b, 10);
+  itoa(DFRobot_MICS_data.Hydrogen_b, DFRobot_MICS_ch.Hydrogen_b, 10);
+  itoa(DFRobot_MICS_data.Hydrothion_b, DFRobot_MICS_ch.Hydrothion_b, 10);
+  itoa(DFRobot_MICS_data.Ammonia_b, DFRobot_MICS_ch.Ammonia_b, 10);
+  itoa(DFRobot_MICS_data.Nitric_Oxide_b, DFRobot_MICS_ch.Nitric_Oxide_b, 10);
+  itoa(DFRobot_MICS_data.Nitrogen_Dioxide_b, DFRobot_MICS_ch.Nitrogen_Dioxide_b, 10);
+
+DFRobot_MICS_data.loop_count = 0;
+DFRobot_MICS_data.Methane = 0.0f;
+DFRobot_MICS_data.Ethanol = 0.0f;
+DFRobot_MICS_data.Hydrogen = 0.0f;
+DFRobot_MICS_data.Ammonia = 0.0f;
+DFRobot_MICS_data.Carbon_Monoxide = 0.0f;
+DFRobot_MICS_data.Nitrogen_Dioxide = 0.0f;
+
+DFRobot_MICS_data.Carbon_Monoxide_b = 0x00;
+DFRobot_MICS_data.Methane_b = 0x00;
+DFRobot_MICS_data.Ethanol_b = 0x00;
+DFRobot_MICS_data.Propane_b = 0x00;
+DFRobot_MICS_data.Iso_Butane_b = 0x00;
+DFRobot_MICS_data.Hydrogen_b = 0x00;
+DFRobot_MICS_data.Hydrothion_b = 0x00;
+DFRobot_MICS_data.Ammonia_b = 0x00;
+DFRobot_MICS_data.Nitric_Oxide_b = 0x00;
+DFRobot_MICS_data.Nitrogen_Dioxide_b = 0x00;
+}
+
+
+void DFRobot_MICS_show(bool json) {
+  convertGasDataToString();
+    if (json) {
+      ResponseAppend_P(PSTR(
+        ",\"DFRobot_MICS\":{"
+        "\"Methane\":\"%s\","
+        "\"Ethanol\":\"%s\","
+        "\"Hydrogen\":\"%s\","
+        "\"Ammonia\":\"%s\","
+        "\"Carbon_Monoxide\":\"%s\","
+        "\"Nitrogen_Dioxide\":\"%s\","
+    
+        "\"Carbon_Monoxide_b\":%s,"
+        "\"Methane_b\":%s,"
+        "\"Ethanol_b\":%s,"
+        "\"Propane_b\":%s,"
+        "\"Iso_Butane_b\":%s,"
+        "\"Hydrogen_b\":%s,"
+        "\"Hydrothion_b\":%s,"
+        "\"Ammonia_b\":%s,"
+        "\"Nitric_Oxide_b\":%s,"
+        "\"Nitrogen_Dioxide_b\":%s"
+        "}"), 
+    
+        DFRobot_MICS_ch.Methane,
+        DFRobot_MICS_ch.Ethanol,
+        DFRobot_MICS_ch.Hydrogen,
+        DFRobot_MICS_ch.Ammonia,
+        DFRobot_MICS_ch.Carbon_Monoxide,
+        DFRobot_MICS_ch.Nitrogen_Dioxide,
+    
+        DFRobot_MICS_ch.Carbon_Monoxide_b,
+        DFRobot_MICS_ch.Methane_b,
+        DFRobot_MICS_ch.Ethanol_b,
+        DFRobot_MICS_ch.Propane_b,
+        DFRobot_MICS_ch.Iso_Butane_b,
+        DFRobot_MICS_ch.Hydrogen_b,
+        DFRobot_MICS_ch.Hydrothion_b,
+        DFRobot_MICS_ch.Ammonia_b,
+        DFRobot_MICS_ch.Nitric_Oxide_b,
+        DFRobot_MICS_ch.Nitrogen_Dioxide_b
+    );
+    }
+}
+
+void DFRobot_MICS_measure(void) {
+
+  float tempFloat;  // Temporary variable for getGasData() (float)
+  int8_t tempInt;   // Temporary variable for getGasExist() (int8_t)
+
+  switch (DFRobot_MICS_data.loop_count) {
+      case 15:
+          tempFloat = mics.getGasData(CH4);
+          if (tempFloat > DFRobot_MICS_data.Methane) DFRobot_MICS_data.Methane = tempFloat;
```

Comment:
Do you expect only to read higher new values? Comparing new value to old value this way will never show lower values....